Treppenhauslichtsteuerung/module.php

```diff
@@ -38,6 +42,25 @@ public function ApplyChanges()
             $this->UnregisterReference($referenceID);
         }
 
+        //Transfer legacy data
+        $transferProperty = function ($legacy, $new)
+        {
+            $newProperty = json_decode($this->ReadPropertyString($new), true);
+            if ($this->ReadPropertyInteger($legacy) != 0) {
+                $newProperty[] = ['VariableID' => $this->ReadPropertyInteger($legacy)];
+                IPS_SetProperty($this->InstanceID, $legacy, 0);
+                IPS_SetProperty($this->InstanceID, $new, json_encode($newProperty));
+                IPS_ApplyChanges($this->InstanceID);
+                return true;
+            }
+        };
+        if ($transferProperty('InputTriggerID', 'InputTriggers')) {
+            return;
+        }
+        if ($transferProperty('OutputID', 'OutputVariables')) {
+            return;
+        }
```

Comment:
Strukturiere das ein bisschen anders, dann kannst du auch beide Anpassungen in einem Durchlauf machen.

Ich persönlich mag es überhaupt nicht, wenn in einem if-Statement Aktionen durchgeführt werden. Ziehe das gerne raus $legacyUpdate = $transferProperty(...) oder dergleichen.